rioxarray/raster_array.py

```diff
@@ -771,11 +772,24 @@ def to_raster(
             This only works if the output raster is tiled. As such, if you
             set this to True, the output raster will be tiled.
             Default is False.
+        compute: bool
+            If True and the data array is a dask array and windowed
```

Comment:
When you say windowed, did you mean `tiled=True`?

Reply:
My understanding from your docstring for `windowed` was that dask isn't used unless the data array is a dask array and windowed=True. Obviously this compute kwarg only comes into play if dask is used so that's what I'm trying to say here.

Reply:
Probably need to fix the wording there. I was trying to say that `windowed` is ignored for dask arrays.

Reply:
Ah ok, so dask.array.store is **always** used if the data is a dask array, right?

Reply:
Currently that is the case. I am working on an implementation that requires lock to be passed in. So, if they don't have a lock passed in, it does not use `dask.array.store`.

Reply:
Darn, I was hoping that using the default `lock=True` for the threaded cases (dask default) would be good enough. Maybe a special section in the documentation for doing multiprocess/cluster work. I guess your way is safer.

Reply:
> I was hoping that using the default lock=True for the threaded cases (dask default) would be good enough.

I run into this issue with the defaultr: https://github.com/dask/distributed/issues/780
Do you have a configuration that works with the default?

Reply:
If you don't create the `Client` object and don't have any other custom configuration then dask *should* should be using the threaded scheduler and the default of `lock=True` will use a thread-safe lock. That worked for me yesterday with this PR. The problem is when using `Client` to create a distributed scheduler (either a local one with multiple processes instead of threads or a multi-node one) then the thread-safe lock doesn't work since it isn't multi-process/node safe. That's what that issue is referring to.

Reply:
Ah, interesting. That is good to know.